src/field/pokemon.ts

```diff
@@ -1350,25 +1348,25 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
   abstract getBossSegmentIndex(): number;
 
   getMoveset(baseOnly?: boolean): PokemonMove[] {
-    const ret = !baseOnly && this.summonData?.moveset ? this.summonData.moveset : this.moveset;
+    const ret = !baseOnly && this.summonData.moveset.length ? this.summonData.moveset : this.moveset;
```

Comment:
NIT: Not length check shortcuts

```suggestion
    const ret = !baseOnly && this.summonData.moveset.length !== 0 ? this.summonData.moveset : this.moveset;
```